lib/package-managers/npm.js

```diff
@@ -159,15 +161,35 @@ module.exports = {
     /**
      * @param {string} packageName
      * @param {string} currentVersion
-     * @param {boolean} pre
+     * @param {{}} options
      * @returns {Promise}
      */
-    latest(packageName, currentVersion, pre) {
+    latest(packageName, currentVersion, options) {
+
+        if (options.enginesNode) {
+            return view(packageName, 'versions-object', currentVersion)
+                .then((versions) => {
+                    const _version = semver.minVersion(options.enginesNode);
+                    if (!_version || !_version.version) {
+                        return versions;
+                    }
+                    return Object.keys(versions).filter((version) => {
+                        let versionEnginesNode = _.get(versions[version], 'engines.node');
+                        if (versionEnginesNode && semver.satisfies(_version.version, versionEnginesNode)) {
+                            return true;
+                        }
+                        return false;
+                    });
+                })
+                .then(filterOutPrereleaseVersions)
+                .then(_.last);
+        }
+
         return view(packageName, 'dist-tags.latest', currentVersion)
```

Comment:
Let's put this section in an `else if` to make it clear that it is exclusive with the `enginesNode` case.